app/models/katello/host/subscription_aspect.rb

```diff
@@ -51,8 +52,33 @@ def self.new_interfaces(facts)
         end
       end
 
+      def self.find_or_create_host(organization, name, rhsm_params)
+        hosts = ::Host.where(:name => name)
+        if hosts.empty? #no host exists
+          Katello::Host::SubscriptionAspect.new_host_from_rhsm_params(rhsm_params, organization, Location.default_location)
+        elsif hosts.where(:organization_id => organization.id).empty? #not in the correct org
+          #TODO
+          fail "Can't handle registering to host in a different org, need to handle this case."
+        else
+          hosts.first
+        end
+      end
+
+      def self.find_or_create_host_for_hypervisor(name, organization, location=nil)
+        location ||= Location.default_location
+        hosts = ::Host.where(:name => name)
+        if hosts.empty? #no host exists
+          ::Host::Managed.new(:name => name, :organization => organization, :location => location, :managed => false)
+        elsif hosts.where(:organization_id => organization.id).empty? #not in the correct org
+          #TODO
+          fail "Can't handle registering to host in a different org, need to handle this case."
+        else
+          hosts.first
+        end
+      end
+
       def update_facts(rhsm_facts)
-        return if self.host.build?
+        return if self.host.nil? || self.host.build?
```

Comment:
I would expect not but I did hit it for the host that is the smart proxy or perhaps from creating hosts via API. I will remove and we can debug further outside of this PR.